File: assignments/a1/q8/Python/Lexspec.py
#Lexspec:top#
#Lexspec:import#

# <lexspec> ::= <rule> <NAME> <QUOTE>
class Lexspec(_Start): #Lexspec:class#

    className = "Lexspec"
    ruleString = "<lexspec> ::= <rule> <NAME> <QUOTE>"
    rule = None
    name = None
    quote = None

    def __init__(rule, name, quote):
        #Lexspec:init#
        self.rule = rule
        self.name = name
        self.quote = quote

#Lexspec#

File: assignments/a1/q8/Java/Lexspec.java
//Lexspec:top//
//Lexspec:import//
import java.util.*;

// <lexspec> ::= <rule> <NAME> <QUOTE>
public class Lexspec extends _Start /*Lexspec:class*/ {

    public static final String $className = "Lexspec";
    public static final String $ruleString =
        "<lexspec> ::= <rule> <NAME> <QUOTE>";

    public Rule rule;
    public Token name;
    public Token quote;

    public Lexspec(Rule rule, Token name, Token quote) {
//Lexspec:init//
        this.rule = rule;
        this.name = name;
        this.quote = quote;
    }

    public static Lexspec parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<lexspec>", scn$.lno);
        Rule rule = Rule.parse(scn$, trace$);
        Token name = scn$.match(Token.Match.NAME, trace$);
        Token quote = scn$.match(Token.Match.QUOTE, trace$);
        return new Lexspec(rule, name, quote);
    }

//Lexspec//
}
